app/controllers/concerns/searchable.rb

```diff
@@ -1,21 +1,30 @@
+# frozen_string_literal: true
+
 # Methods pertaining to searching for objects in a controller action.
 module Searchable
   extend ActiveSupport::Concern
 
+  # Builds a searcher with which to search.
+  def build_searcher(searcher_class, relation, query, scope, **options)
+    searcher_class.new(relation, query, scope, options)
+  end
+
   # If params[:search] is present, runs a search of `klass`,
-  # passing `rel` as the relation to which to apply the search.
+  # passing `relation` as the relation to which to apply the search,
+  # and the current_mission as the search scope.
+  #
   # Returns the new relation if search succeeds,
-  # otherwise sets flash, flash[:search_error] = true, and returns `rel` unchanged.
-  def apply_search_if_given(klass, rel, **options)
-    params[:search].present? ? apply_search(klass, rel, **options) : rel
-  end
+  # otherwise sets flash, flash[:search_error] = true, and returns `relation` unchanged.
+  def apply_search(searcher_class, relation, **options)
+    query = params[:search]
+    return relation if query.blank?
```

Comment:
Maybe use case statement here or in build_searcher to get Searcher class from relation (`relation.klass`)